### DISCUSSION TITLE
What's the problem of this method to transpile from es5 to 6 for tree shaking?

### DISCUSSION THREAD

**Discussion by nyngwang** [babel/babel#14447] [Q&A] [answered]
I just found some article that uses the following method to transpile CJS require to ESM import for tree-shaking:

```js
"plugins": [
    [
      "babel-plugin-import",
      {
        "libraryName": "@material-ui/core",
        "libraryDirectory": "/",
        "camel2DashComponentName": false
      },
      "core"
    ],
```

My question is simple: How do you think about this method? Does it cause any trouble if one uses it without care?

**Comment by nicolo-ribaudo** [comment, marked as answer]
Looking at https://github.com/umijs/babel-plugin-import, it looks like it compiled ESM import to CJS require, not CJS require to ESM import :thinking:

**Reply by nyngwang** [reply]
Thank you... I was misled by an answer on StackOverflow :(
